pkg/service/keystore/storage/bolt.go

```diff
@@ -50,6 +50,33 @@ func (b BoltKeyStoreStorage) storeServiceKey(key ServiceKey) error {
 	return nil
 }
 
+// getServiceKey attempts to get the service key from memory, and if not available rehydrates it from the DB
+func (b BoltKeyStoreStorage) getServiceKey() ([]byte, error) {
+	if len(b.serviceKey) != 0 {
+		return b.serviceKey, nil
```

Comment:
it will always return here right? NewBoltKeyStoreStorage sets the service key and it should never be unset?

Is this like if someone spins up the service with an empty service key set by default it will load from db?

Reply:
yeah, exactly. someone can spin/up down the service and have it point to the same storage this is a way to rehydrate the memory representation of the key.